TimoCloud-Universal/src/main/java/cloud/timo/TimoCloud/base/managers/BaseInstanceManager.java

```diff
@@ -456,21 +452,23 @@ private void startProxy(BaseProxyObject proxy) {
         }
     }
 
-    private Integer getFreePort(int offset) {
-        for (int p = offset; p <= offset + 1000; p++) {
-            if (portIsFree(p)) return p;
-        }
-        return null;
-    }
+    private Integer getFreePortCommon(int startPort, int currentPort, int maxPort) throws CommonStartException {
+        Integer freePort = null;
 
-    private void blockPort(int port) {
-        recentlyUsedPorts.put(port, 60);
+        for (int p = currentPort; p <= maxPort; p++)
+            if (portIsFree(p)) freePort = p;
```

Comment:
There are some problems with this method. Firstly, we want the loop to break when a free port is found. Secondly, we want to continue the loop when `maxPort` is reached, but have a maximum of `maxPort-startPort+1`.

Hence, we need something like:
```
for (int i = 0; i<=maxPort-startPort; i++) {
    int port = (currentPort + i) % maxPort;
    if (port < startPort) port += startPort;
    if (isPortFree(p)) return p;
}
throw new InstanceStartException("No free port found. Please report this!");
```